grid/chat/members/maps/slack.suma

```diff
@@ -0,0 +1,187 @@
+profile = "chat/members@1.0"
+provider = "slack"
+
+"""
+GetMembers map
+"""
+map GetMembers {
+  http GET "/users.list" {
+    request "application/x-www-form-urlencoded" {
+      query {
+        cursor = input.page
+        limit = input.limit || 100
+      }
+
+      headers {
+        "Authorization" = `Bearer ${parameters.accessToken}`
+      }
+    }
+
+    response 200 "application/json" {
+      error = call MapSlackError(
+        statusCode = statusCode,
+        error = body,
+        headers = headers,
+      ) if (!body.ok)
+
+      return map error if (!body.ok) error
+
+      members = call MapMembers(members = body.members)
+      rateLimit = call MapRateLimitDetails(headers = headers)
+      result.members = members
+
+      set if (body.response_metadata && body.response_metadata.next_cursor) {
+        result.nextPage = body.response_metadata.next_cursor
+      }
+
+      set if (rateLimit) {
+        result.rateLimit = rateLimit
+      }
+
+      map result result
+    }
+
+    response {
+      error = call MapSlackError(
+        statusCode = statusCode,
+        error = body,
+        headers = headers,
+      )
+      
+      map error error
+    }
+  }
+}
+
+"""
+Operation MapMembers
+Returns array mapping members in received body
+"""
+operation MapMembers {
+  members = args.members.map(member => {
+    const resultMember = {
+      id: member.id,
+      isBot: member.is_bot,
+    }
+
+    if (member.profile && member.profile.display_name) {
+      resultMember.username = member.profile.display_name;
+    }
+
+    if (member.name && resultMember.username === undefined) {
```

Comment:
So, technically, can I end up with empty username on Slack?

Reply:
I wasn't sure if profile display name is not optional based on slack docs, but it's among common fields [here](https://api.slack.com/types/user), so I'll remove the checks.